stackslib/src/chainstate/nakamoto/coordinator/mod.rs

```diff
@@ -1057,7 +1089,34 @@ impl<
             let reward_cycle_info = if self.burnchain.is_reward_cycle_start(header.block_height) {
                 // we're at the end of the prepare phase, so we'd better have obtained the reward
                 // cycle info of we must block.
-                let reward_cycle_info = self.get_nakamoto_reward_cycle_info(header.block_height)?;
+                // NOTE(safety): the reason it's safe to use the local best stacks tip here is
+                // because as long as at least 30% of the signers are honest, there's no way there
+                // can be two or more distinct reward sets calculated for a reward cycle.  Due to
+                // signature malleability, there can be multiple unconfirmed siblings at a given
+                // height H, but at height H+1, exactly one of those siblings will be canonical,
+                // and will remain canonical with respect to its tenure's Bitcoin fork forever.
+                // Here, we're loading a reward set calculated between H and H+99 from H+100, where
+                // H is the start of the prepare phase.  So if we get any reward set from our
+                // canonical tip, it's guaranteed to be te canonical one.
+                let canonical_sortition_tip = self.canonical_sortition_tip.clone().unwrap_or(
+                    // should be unreachable
```

Comment:
If this should be unreachable, should we log a warning here?

Reply:
In the event that this code does get reached, we have a recovery path -- load the canonical sortition tip from the DB.  The chains coordinator ought to always set `self.canonical_sortition_tip`, but in this case, there's a way to recover if it doesn't for some reason.